pkg/steps/release/promote.go

```diff
@@ -64,18 +70,58 @@ func (s *promotionStep) run(ctx context.Context) error {
 		return fmt.Errorf("could not resolve pipeline imagestream: %w", err)
 	}
 
-	imageMirrorTarget := getImageMirrorTarget(tags, pipeline, registryDomain(s.configuration.PromotionConfiguration))
+	imageMirrorTarget, namespaces := getImageMirrorTarget(tags, pipeline, registryDomain(s.configuration.PromotionConfiguration))
 	if len(imageMirrorTarget) == 0 {
 		logrus.Info("Nothing to promote, skipping...")
 		return nil
 	}
 
+	if err := s.ensureNamespaces(ctx, namespaces); err != nil {
+		return fmt.Errorf("failed to ensure namespaces to promote to in central rewgistry: %w", err)
+	}
+
 	if _, err := steps.RunPod(ctx, s.client, getPromotionPod(imageMirrorTarget, s.jobSpec.Namespace())); err != nil {
 		return fmt.Errorf("unable to run promotion pod: %w", err)
 	}
 	return nil
 }
 
+func (s *promotionStep) ensureNamespaces(ctx context.Context, namespaces sets.String) error {
+	var dockercfg credentialprovider.DockerConfigJSON
+	if err := json.Unmarshal(s.pushSecret.Data[coreapi.DockerConfigJsonKey], &dockercfg); err != nil {
+		return fmt.Errorf("failed to deserialize push secret: %w", err)
+	}
+
+	appCIDockercfg, hasAppCIDockercfg := dockercfg.Auths[api.ServiceDomainAPPCIRegistry]
+	if !hasAppCIDockercfg {
+		return fmt.Errorf("push secret has no entry for %s", api.ServiceDomainAPPCIRegistry)
+	}
+
+	appCIKubeconfig := &rest.Config{Host: "https://api.ci.l2s4.p1.openshiftapps.com:6443", BearerToken: appCIDockercfg.Password}
```

Comment:
Do we have this as a constant somewhere?

Reply:
Nope

Reply:
Can we add one?

Reply:
Done